src/casts/convert-closures.rkt

```diff
@@ -1974,129 +1964,129 @@ TODO We can generate better code in this pass for function casts.
      (convert-closures
       '()
       (Let
-          (list
-           (cons init (Unguarded-Box (Quote #f)))
-           (cons loop4 (Unguarded-Box (Quote 0))))
-        (Letrec
+       (list
+        (cons init (Unguarded-Box (Quote #f)))
+        (cons loop4 (Unguarded-Box (Quote 0))))
+       (Letrec
+        (list
+         (cons
+          loop5
+          (Lambda
+           '()
+           (Castable
+            fn-cast
+            (App-Code
+             app-crcn-cl
+             (list
+              (Begin
+                (list
+                 (App-Fn-or-Proxy apply-coercion (Var loop5) '()))
+                (App-Fn-or-Proxy
+                 apply-coercion
+                 (Unguarded-Box-Ref (Var loop4)) '()))
+              (Quote-Coercion (Static-Id (Uid "static_project_crcn" 464)))
+              (Quote 0) (Quote 0) (Quote 0)))))))
```

Comment:
These 3 zeros input to the `apply-coercion` function are no longer needed I believe. I think they should be replaced with a single occurrence of `do-not-suspend-monotonic-heap-casts`. This is my fault.